File: learn-go/challenges/reverse_integer.go
package main

import "fmt"

func reverseGivenInteger(num int) int {
	reservedNum := 0
	for num > 0 {
		remainder := num % 10
		reservedNum *= 10
		reservedNum += remainder
		num /= 10
	}
	return reservedNum
}

func main() {
	var num = 3849
	fmt.Printf("Reversed integer is: %d\n", reverseGivenInteger(num))
	num = 2222
	fmt.Printf("Reversed integer is: %d\n", reverseGivenInteger(num))
	num = 1020
	fmt.Printf("Reversed integer is: %d\n", reverseGivenInteger(num))
	num = 6031
	fmt.Printf("Reversed integer is: %d\n", reverseGivenInteger(num))
}

File: learn-js/exercise/reverse-integer.js
const num = 3849;

function reverseGivenInteger(num) {
  let reverseNum = 0
  let lastDigit
  while (num !== 0) {
    lastDigit = num % 10
    reverseNum = reverseNum*10 + lastDigit
    num = Math.floor(num / 10)
  }

  return reverseNum;
}

console.log(`Reversed integer is: ${reverseGivenInteger(num)}`)
console.log(`Reversed integer is: ${reverseGivenInteger(1111)}`)
console.log(`Reversed integer is: ${reverseGivenInteger(64684)}`)
console.log(`Reversed integer is: ${reverseGivenInteger(86173)}`)